contracts/YieldStEthLever.sol

```diff
@@ -0,0 +1,467 @@
+        // borrowed amount later pays off the flash loan. This makes sure we
+        // borrow exactly `borrowAmount`.
+        IPool pool;
+        {
+            uint128 netInvestAmount = uint128(baseAmount + borrowAmount - fee);
+
+            // Get WEth by selling borrowed FYTokens. We don't need to check for a
+            // minimum since we check that we have enough collateral later on.
+            pool = IPool(ladle.pools(seriesId));
+            IFYToken fyToken = pool.fyToken();
+            fyToken.safeTransfer(address(pool), netInvestAmount);
+        }
+        uint256 wethReceived = pool.sellFYToken(address(this), 0);
+
+        // Swap WEth for StEth on Curve.fi. Again, we do not check for a
+        // minimum.
+        // 0: WEth
+        // 1: StEth
+        uint256 boughtStEth = stableSwap.exchange(
+            0,
+            1,
+            wethReceived,
+            0,
+            address(this)
+        );
+
+        // Wrap StEth to WStEth.
+        uint128 wrappedStEth = uint128(wsteth.wrap(boughtStEth));
+
+        // This is the amount to deposit, so we check for slippage here. As
+        // long as we end up with the desired amount, it doesn't matter what
+        // slippage occurred where.
+        if (wrappedStEth < minCollateral) revert SlippageFailure();
+
+        // Deposit WStEth in the vault & borrow `borrowAmount` fyToken to
+        // pay back.
+        wsteth.safeTransfer(address(wstethJoin), wrappedStEth);
+        ladle.pour(
+            vaultId,
+            address(this),
+            int128(uint128(wrappedStEth)),
+            int128(uint128(borrowAmount))
+        );
+
+        // At the end, the flash loan will take exactly `borrowedAmount + fee`,
+        // so the final balance should be exactly 0.
+    }
+
+    /// @notice Unwind a position.
+    ///
+    ///     If pre maturity, borrow liquidity tokens to repay `art` debt and
+    ///     take `ink` collateral. Repay the loan and return remaining
+    ///     collateral as WEth.
+    ///
+    ///     If post maturity, borrow WEth to pay off the debt directly. Convert
+    ///     the WStEth collateral to WEth and return excess to user.
+    ///
+    ///     This function will take the vault from you using `Giver`, so make
+    ///     sure you have given it permission to do that.
+    /// @param ink The amount of collateral to recover.
+    /// @param art The debt to repay.
+    /// @param minWeth Revert the transaction if we don't obtain at least this
+    ///     much WEth at the end of the operation.
+    /// @param vaultId The vault to use.
+    /// @param seriesId The seriesId corresponding to the vault.
+    /// @dev It is more gas efficient to let the user supply the `seriesId`,
+    ///     but it should match the pool.
+    function unwind(
+        uint128 ink,
+        uint128 art,
+        uint256 minWeth,
+        bytes12 vaultId,
+        bytes6 seriesId
```

Comment:
Parameters should be ordered:

    Identifiers
    Amounts
    Conditionals
```
        bytes12 vaultId,
        bytes6 seriesId,
        uint128 ink,
        uint128 art,
        uint256 minWeth
```

Maybe here we should use something other than `ink` and `art`, since we know that they are wstEth and weth respectively. `wethLoan` and `wstEthWithdrawn` or something like that?